### DISCUSSION TITLE
Add the trace-id to the response headers

### DISCUSSION THREAD

**Discussion by ziomill** [open-telemetry/opentelemetry-java-instrumentation#7495] [Q&A]
Hi all,
I'm trying OpenTelemetry auto instrumentation to handle traces on a Java Spring Boot project (that exposes Rest APIs and GRPC services).

It all seems to work as expected, but I would have expected that the open telemetry agent would have returned the trace-id in the headers. 

Is there a configuration to enable in order to add of this kind of information to the response or is there a reason why this not happens? 

Thank you

**Comment by mateuszrzeszutek** [comment]
Hey @ziomill ,

There is already an open issue for this: https://github.com/open-telemetry/opentelemetry-java-instrumentation/issues/5353

In general, I believe that this is a feature that we'd like to have in the javaagent (most likely with a very configurable header(s) format; and also disabled by default). However, the main problem with it is that currently there is no spec for the HTTP response header that would contain the server trace id. Some companies (like Splunk) use the `Server-Timing` header, which kinda sorta fits the purpose, but it definitely is not the "official" specced solution.

**Reply by ziomill** [reply]
Thank you @mateuszrzeszutek , 
I understand the problem and I will follow the issue that you have linked.
In the meantime, I will implement a custom interceptor that will add the trace-id to each response in a custom header.
